### DISCUSSION TITLE
parsing dash '-' in json path

### DISCUSSION THREAD

**Discussion by romange** [danielaparker/jsoncons#472] [Q&A]
Seems that jsoncons rejects '-' when it appears in field names. Is it by design? I know it's still possible to use quotes for referencing fields but unfortunately we get our jsonpath expression from a user that uses dashes and preprocessing the string before passing it further to `jsonpath::make_expression` is complicated.

**Comment by danielaparker** [comment]
By design, see [jsoncons JSONPath](https://danielaparker.github.io/JsonCons.Net/articles/JsonPath/JsonConsJsonPath.html). Allowing hyphens in bare property names conflicts with allowing unary minus and subtraction in filter expressions.
